### PR TITLE
[youtube] misc auth cleanup

### PR DESCRIPTION
**Pull request by coletdjnz**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [X] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [X] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [X] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [X] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [X] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.

Just some minor cleanup.

changes:
 - deprecate identity_token: doesn't appear to be used by YouTube anymore (at least in most main calls). I've removed it from function parameters but kept it being extracted from ytcfg in the generate api headers func for now.
 - Refactored generate api headers func (including parameter order) - think this makes it easier to read now.
 - add `is_authenticated` for readability
 - other misc cleanup

### CODE REVIEW

**Comment by coletdjnz** [comment]
I've gone through this thoroughly a few times now to make sure I didn't break anything by changing the parameter order, so should be good.

But another pair of eyes doesn't hurt

**Comment by pukkandan** [comment]
> I've gone through this thoroughly a few times now to make sure I didn't break anything by changing the parameter order

For functions like generate_api_headers that have a ton of params and there is no intuitive order, use `*` to enforce kwargs style usage

File: yt_dlp/extractor/youtube.py [hunk 1]
```diff
@@ -2409,18 +2412,10 @@ def _generate_comment_continuation(video_id):
...
-                    for key, renderer in entry.items():
-                        if key not in known_entry_comment_renderers:
-                            continue
-                        yield from self._comment_entries(
-                            renderer, video_id=video_id, ytcfg=ytcfg,
-                            identity_token=self._extract_identity_token(webpage, item_id=video_id),
-                            account_syncid=self._extract_account_syncid(ytcfg))
-                        break
+            yield from self._comment_entries(
+                traverse_obj(contents, (..., ('itemSectionRenderer',)), get_all=False), ytcfg, video_id)
```
Comment (pukkandan) [inline yt_dlp/extractor/youtube.py:2416]: the extra tuple isn't needed, right?
```suggestion
                traverse_obj(contents, (..., 'itemSectionRenderer'), get_all=False), ytcfg, video_id)
```

File: yt_dlp/extractor/youtube.py [hunk 2]
```diff
@@ -2445,19 +2440,23 @@ def _real_comment_extract(contents):
...
+        playback_context = {
             'playbackContext': {
                 'contentPlaybackContext': context
-            },
-            'contentCheckOk': True,
-            'racyCheckOk': True
+            }
         }
+        playback_context.update(cls._get_checkok_params())
+        return playback_context
```
Comment (pukkandan) [inline yt_dlp/extractor/youtube.py:2459]:
```suggestion
        return {
            'playbackContext': {
                'contentPlaybackContext': context
            },
            **cls._get_checkok_params()
        }
```
nothing wrong with your code, just a suggestion

**Review by pukkandan — APPROVED** [review]

**Pull request merged**